specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensionTypes.json

```diff
@@ -0,0 +1,424 @@
+          },
+          {
+            "$ref": "#/parameters/ClusterNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/ExtensionTypeNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/ExtensionType"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{clusterRp}/{clusterType}/{clusterName}/providers/Microsoft.KubernetesConfiguration/extensionTypes": {
+      "get": {
+        "description": "Get Extension Types",
+        "operationId": "ClusterExtensionTypes_List",
+        "x-ms-examples": {
+          "ClusterExtensionTypes_List": {
+            "$ref": "./examples/ClusterExtensionTypes_List.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ClusterRpParameter"
+          },
+          {
+            "$ref": "#/parameters/ClusterTypeParameter"
+          },
+          {
+            "$ref": "#/parameters/ClusterNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/ExtensionType"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/{clusterRp}/{clusterType}/locations/{location}/extensionTypes/{extensionTypeName}/versions": {
+      "get": {
+        "description": "List available versions for an Extension Type",
+        "operationId": "ExtensionTypeVersions_List",
+        "x-ms-examples": {
+          "ExtensionTypeVersions_List": {
```

Comment:
I'm not understanding this one... if clusterRp is not the kubernetesconfiguration rp, how does kubernetesconfiguration own the extension type/versions?

Reply:
did you mean to write /subscriptions/{subscriptionId}/providers/{clusterRp}/{clusterType}/locations/{location}/providers/Microsoft.KubernetesConfiguration/extensionTypes/{extensionTypeName}" or something similar?

---
In reply to: [608965251](https://github.com/Azure/azure-rest-api-specs/pull/13562#discussion_r608965251) [](ancestors = 608965251)

Reply:
the url is missing multiple segments like the clusterName and providers/Microsoft.KubernetesConfiguration

---
In reply to: [608965251](https://github.com/Azure/azure-rest-api-specs/pull/13562#discussion_r608965251) [](ancestors = 608965251)